template-for-repository/proofs/Makefile.common

```diff
@@ -300,54 +273,137 @@ CBMC_REMOVE_FUNCTION_BODY := $(patsubst %,--remove-function-body %, $(REMOVE_FUN
 
 # Compile project sources
 $(PROJECT_GOTO)1.goto: $(PROJECT_SOURCES)
-	mkdir -p $(dir $@)
-	mkdir -p $(dir $(call LOG_FROM_ENTRY,$@))
-	$(call DO_GOTO_CC,--export-file-local-symbols $(COMPILE_FLAGS) $(INCLUDES) $(DEFINES),$^,$@)
+	$(LITANI) add-job \
+	  --command \
+	    '$(GOTO_CC) $(CBMC_VERBOSITY) --export-file-local-symbols $(COMPILE_FLAGS) $(INCLUDES) $(DEFINES) $^ -o $@' \
+	  --inputs $^ \
+	  --outputs $@ $(call LOG_FROM_ENTRY,$@) \
+	  --pipeline-name $(HARNESS_ENTRY) \
+	  --ci-stage build \
+	  --interleave-stdout-stderr \
+	  --stdout-file $(call LOG_FROM_ENTRY,$@) \
+	  --description "$(HARNESS_ENTRY): building project binary"
 
 # Compile proof sources
 $(PROOF_GOTO)1.goto: $(PROOF_SOURCES)
-	mkdir -p $(dir $@)
-	mkdir -p $(dir $(call LOG_FROM_ENTRY,$@))
-	$(call DO_GOTO_CC,--export-file-local-symbols $(COMPILE_FLAGS) $(INCLUDES) $(DEFINES),$^,$@)
+	$(LITANI) add-job \
```

Comment:
A lot of these calls have the same set of flags over and over. I wonder if there is a cleaner way to wrap that. Maybe a "standard_flags" set?

Reply:
I thought about this, and in fact I originally used the macros that you had implemented (which are deleted above). In the end I erred on the side of 'explicit is better than implicit' and avoiding too much redirection, since Litani is an unfamiliar tool. Really the only two flags that are common across all jobs are `--inputs $^ --outputs $@ $(call LOG_FROM_ENTRY,$@)`, and there are a few more flags that are on most but not all jobs.

Up to you, I don't mind either reintroducing the macros or having variables, but this way read the most intuitively in my opinion.